### ISSUE TITLE
NeighborLoader with isolated nodes

### ISSUE THREAD

**Issue by RX28666** [pyg-team/pytorch_geometric#5846]
### 🐛 Describe the bug

Hello, I tried to use NeighborLoader on some datasets with isolated nodes, and then use `to_scipy_sparse_matrix` to covert edge_index to adj matrix, like following:
```
data = dataset[0]
data.edge_index = to_undirected(data.edge_index)
data = data.to(device)
data.n_id = torch.arange(data.num_nodes)
kwargs = {'batch_size': 3327, 'num_workers': 0, 'persistent_workers': False}
train_loader = NeighborLoader(data, input_nodes=None,
                              num_neighbors=[-1], shuffle=True, **kwargs)
for i in range(500):
    for batch in train_loader:
        print(torch_geometric.utils.to_scipy_sparse_matrix(batch.edge_index).shape)
```
I am using Citeseer, I put all 3327 nodes into one batch for verifying firstly, but the output of the above code not always 3327, sometimes becomes 3326, like following:
```
(3327, 3327)
(3327, 3327)
(3327, 3327)
(3327, 3327)
(3327, 3327)
(3326, 3326)
(3327, 3327)
(3327, 3327)
(3327, 3327)
```
I tried the method you mentioned in #5827, but this problem still happened. 
I was wondering if If it's because of isolated nodes, but sometimes the result is correct(3327), sometimes is wrong(3326).
Any help would be appreciated.

### Environment

* PyG version:2.1.0
* PyTorch version:1.12
* OS: colab
* Python version:3
* CUDA/cuDNN version:11.3
* How you installed PyTorch and PyG (`conda`, `pip`, source):pip
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
Yes, this is due to isolated nodes and `shuffle=True`. This will lead to isolated nodes being place last in the batch, and since there are no edges, `to_scipy_sparse_matrix` will produce a minimal sized matrix to fit the graph. You can specify `num_nodes` in the `to_scipy_sparse_matrix` function to fix this.
